src/service.rs

```diff
@@ -1493,47 +1493,51 @@ where
     }
 
     #[inline]
-    fn user_task_poll(&mut self, cx: &mut Context) {
-        loop {
-            if self.write_buf.len() > self.config.session_config.send_event_size()
-                && self.high_write_buf.len() > self.config.session_config.send_event_size()
-            {
-                break;
-            }
+    fn user_task_poll(&mut self, cx: &mut Context) -> Poll<Option<()>> {
+        if self.write_buf.len() > self.config.session_config.send_event_size()
+            && self.high_write_buf.len() > self.config.session_config.send_event_size()
+        {
+            return Poll::Pending;
```

Comment:
It's better to add a comment here to explain who is responsible to notify this `user_task_poll` since it does not depend on a child poll.